### PR TITLE
Delete SimplePromise and use regular native Promise.

### PR DESCRIPTION
**Pull request by rniwa**
Also deleted the unused _waitAndWarmUp.

### CODE REVIEW

**Review by shvaikalesh — APPROVED** [review]

File: resources/benchmark-runner.js [hunk 1]
```diff
@@ -95,33 +61,12 @@ BenchmarkRunner.prototype._appendFrame = function (src) {
...
-    return promise;
-}
-
 BenchmarkRunner.prototype._writeMark = function(name) {
     if (window.performance && window.performance.mark)
         window.performance.mark(name);
 }
 
-// This function ought be as simple as possible. Don't even use SimplePromise.
+// This function ought be as simple as possible. Don't even use Promise.
```
Comment (smaug----) [inline resources/benchmark-runner.js:69]: (random comment, the setTimeout call there isn't "simple". An implementation may or may not call rAF before the timeout runs. And rAF can be rather heavy operation. So the results can be a bit surprising.  But not about this patch :) )

Comment (rniwa) [reply resources/benchmark-runner.js:69]: Yeah, I think we want to refine this stuff in v3. I'm observing odd behavior for https://github.com/WebKit/Speedometer/pull/11 where async time sometimes disappears.

**Review by smaug---- — APPROVED** [review]
This of course does change a bit how tests are run, since real promises use microtask scheduling, but anyhow, we should do this.

**Review by kyliau — APPROVED** [review]

**Comment by rniwa** [comment]
Thanks for the review, everyone.

**Pull request merged**